src/components/color-picker/color-field/color-field.tsx

```diff
@@ -0,0 +1,494 @@
+      this.#canvasRef.width = this.#canvasRef.clientWidth;
+      this.#canvasRef.height = this.#canvasRef.clientHeight;
+      this.#drawColorField();
+    }
+  };
+
+  #revertToOldValue = (oldValue: string): void => {
+    const fallbackValue = oldValue || "#000";
+
+    this.#isUserAction = true;
+    this.value = fallbackValue;
+  };
+
+  #updateMarkerPosition = (): void => {
+    if (!this.colorVariants || !this.#canvasRef) {
+      return;
+    }
+
+    // Use HSV for marker positioning
+    const hsvValues = this.colorVariants.hsv;
+    // X represents saturation (0-100%)
+    this.#currentX = (hsvValues.s / 100) * this.#canvasRef.width;
+    // Y represents value (0% value = bottom, 100% value = top)
+    this.#currentY = ((100 - hsvValues.v) / 100) * this.#canvasRef.height;
+    this.#drawMarker();
+  };
+
+  #updateCoordinates = (event: MouseEvent): void => {
+    const rect = this.#canvasRef.getBoundingClientRect();
+    const canvasHeight = this.#canvasRef.height;
+    const canvasWidth = this.#canvasRef.width;
+
+    // Ensure valid pixel range
+    this.#currentX = Math.max(
+      0,
+      Math.min(canvasWidth, event.clientX - rect.left)
+    );
+    this.#currentY = Math.max(
+      0,
+      Math.min(canvasHeight, event.clientY - rect.top)
+    );
+
+    this.#colorPickedVariants = this.#getColorFromCanvas();
+    this.#drawMarker();
+  };
+
+  componentWillLoad() {
+    // Initialize colorVariants before component loads to avoid state change on load
+    if (this.value) {
+      try {
+        this.colorVariants = fromStringToColorVariants(this.value);
+        const format = getColorFormat(this.value);
+
+        if (format) {
+          this.#colorFormat = format;
+        }
+      } catch (error) {
+        console.warn(`Invalid value color: ${this.value}`, error);
+        this.colorVariants = fromStringToColorVariants("#000000");
+        this.#colorFormat = "hex";
+      }
+    }
+  }
+
+  connectedCallback() {
+    // Accessibility
+    this.internals.setFormValue(this.value);
+    const labels = this.internals.labels;
+    const accessibleNameFromExternalLabel = getElementInternalsLabel(labels);
+
+    // Report any accessibility issue
+    analyzeLabelExistence(
+      this.el,
+      "ch-color-field",
+      labels,
+      accessibleNameFromExternalLabel,
+      this.translations.accessibleName.label
+    );
+  }
+
+  componentDidLoad() {
+    this.#updateCanvasSize();
+    this.#updateMarkerPosition();
+
+    // Initialize ResizeObserver with RAF optimization
+    this.#resizeObserver = new ResizeObserver(() => {
+      this.#resizeRAF.perform(() => {
+        this.#updateCanvasSize();
+        this.#updateMarkerPosition();
+      });
+    });
+
+    // Observe the canvas for size changes
+    this.#resizeObserver.observe(this.#canvasRef);
+  }
+
+  disconnectedCallback() {
+    // Disconnect the observer when the component is unloaded
+    if (this.#resizeObserver) {
+      this.#resizeObserver.disconnect();
```

Comment:
And clean up the reference

```suggestion
      this.#resizeObserver.disconnect();
      this.#resizeObserver = undefined;
```